lib/glific/communications/message.ex

```diff
@@ -441,15 +442,26 @@ defmodule Glific.Communications.Message do
 
   defp get_received_msg_publish_event(_), do: :received_message
 
-  @spec get_group_id(map()) :: non_neg_integer() | nil
-  defp get_group_id(%{provider: "maytapi"} = message_params) do
-    with %Group{id: id} <-
-           Repo.get_by(Group, %{bsp_id: message_params.group_id},
-             organization_id: message_params.organization_id
-           ) do
-      id
+  @spec create_or_get_group_id(map()) :: non_neg_integer() | nil
```

Comment:
since this is function will be specific to maytapi/whatsapp group so make it public function in `lib/glific/groups/whatsapp_group.ex`